### ISSUE TITLE
Live ws does not work when grafana is in iframe and run behind reverse proxy

### ISSUE THREAD

**Issue by simPod** [grafana/grafana#36291]
**What happened**:

I'm not sure if it's a bug or not but I'm using grafana within iframe and this PR made it stop working #35983.

This check

https://github.com/grafana/grafana/blob/6f38883583c4c43af149f68db482b39a3240ec95/pkg/services/live/live.go#L384-L387

fails because origin and app url are different

`origin=https://site.with.iframe.com appUrl=http://localhost:3000/grafana/`

**How to reproduce it (as minimally and precisely as possible)**:

```nginx
server {
        location /grafana {
                proxy_http_version 1.1;
                proxy_read_timeout 7d;
                proxy_set_header Host $http_host;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection "upgrade";
                proxy_pass http://localhost:3000;
        }
}
```

```ini
[server]
root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana
serve_from_sub_path = true
```

**Environment**:
- Grafana version: master (6f38883583c4c43af149f68db482b39a3240ec95)

**Comment by dsotirakis** [comment]
Hello @simPod , thanks for raising the issue, I can see that you've already raised a PR for your forked repo, would it be okay for you to raise a PR in our repo too? So we can properly review, test and discuss? Only if you feel like it :)

**Comment by FZambia** [comment]
Fixed by #36318
